fdb-record-layer-core/src/test/java/com/apple/foundationdb/record/provider/foundationdb/FDBStoreTimerTest.java

```diff
@@ -351,15 +352,117 @@ void testTransactionMetricListener() {
                 }
             }
             assertThat(listener.transactions, equalTo(3));
-            assertThat(listener.reads, equalTo(timer.getCount(FDBStoreTimer.Counts.READS)));
-            assertThat(listener.writes, equalTo(timer.getCount(FDBStoreTimer.Counts.WRITES)));
+            assertThat(listener.reads, equalTo(6));
+            assertThat(listener.writes, equalTo(2));
             assertThat(listener.commits, equalTo(2));
             assertThat(listener.closes, equalTo(3));
         } finally {
             FDBDatabaseFactory.instance().setTransactionListener(null);
         }
     }
 
+    @Test
+    void testDelayedForCommit() {
+        final FDBStoreTimer timer = new FDBStoreTimer();
+        try (FDBRecordContext context = fdb.openContext(null, timer)) {
+            Transaction tr = context.ensureActive();
+            tr.clear(subspace.range());
+            assertEquals(0, timer.getCount(FDBStoreTimer.Counts.RANGE_DELETES));
+            context.commit();
+            assertEquals(1, timer.getCount(FDBStoreTimer.Counts.RANGE_DELETES));
+        }
+    }
+
+    @Test
+    void testMultipleTransactionsShareTimerSomeCommit() {
+        final int contextCount = 20;
+        final FDBStoreTimer timer = new FDBStoreTimer();
+        final List<FDBRecordContext> contexts = new ArrayList<>(contextCount);
+        try {
+            for (int i = 0; i < contextCount; i++) {
+                FDBRecordContext context = fdb.openContext(null, timer);
+                context.getReadVersion();
+                contexts.add(context);
+            }
+            assertEquals(contextCount, timer.getCount(FDBStoreTimer.Events.GET_READ_VERSION));
+
+            int bytesRead = 0;
+            int bytesWritten = 0;
+            for (int i = 0; i < contextCount; i++) {
+                FDBRecordContext context = contexts.get(i);
+                byte[] key = subspace.pack(Tuple.from(i / 5, i % 5));
+                byte[] value = context.ensureActive().get(key).join();
+                context.ensureActive().set(key, value);
+
+                bytesRead += value.length;
+                if (i % 2 == 0) {
+                    bytesWritten += key.length + value.length;
+                }
+            }
+            assertEquals(contextCount, timer.getCount(FDBStoreTimer.Counts.READS));
+            assertEquals(bytesRead, timer.getCount(FDBStoreTimer.Counts.BYTES_READ));
+            assertEquals(0, timer.getCount(FDBStoreTimer.Counts.WRITES));
+            assertEquals(0, timer.getCount(FDBStoreTimer.Counts.BYTES_WRITTEN));
+
+            // Only commit the event transactions (note: i += 2, not i++)
```

Comment:
I think this is what you meant to say
```suggestion
            // Only commit half the transactions (note: i += 2, not i++)
```